providers/oidc.go

```diff
@@ -194,15 +232,31 @@ func getOIDCHeader(accessToken string) http.Header {
 	return header
 }
 
-func findClaimsFromIDToken(idToken *oidc.IDToken, accessToken string, profileURL string) (*OIDCClaims, error) {
+func (p *OIDCProvider) findClaimsFromIDToken(idToken *oidc.IDToken, accessToken string, profileURL string) (*OIDCClaims, error) {
 
 	// Extract custom claims.
 	claims := &OIDCClaims{}
-	if err := idToken.Claims(claims); err != nil {
+	if err := idToken.Claims(&claims.rawClaims); err != nil {
 		return nil, fmt.Errorf("failed to parse id_token claims: %v", err)
 	}
 
-	if claims.Email == "" {
+	// Extract custom claims manually (to avoid parsing twice)
+	userID := claims.rawClaims[p.UserIDClaim]
+	if userID == nil {
+		return nil, fmt.Errorf("claims did not contains the required user-id-claim '%s'; claims: %v", p.UserIDClaim, claims.rawClaims)
+	}
+	claims.UserID = fmt.Sprint(userID)
+	if value, ok := claims.rawClaims["sub"].(string); ok {
+		claims.Subject = value
+	}
+	if value, ok := claims.rawClaims["email_verified"].(*bool); ok {
+		claims.Verified = value
+	}
+	if value, ok := claims.rawClaims["preferred_username"].(string); ok {
+		claims.PreferredUsername = value
+	}
```

Comment:
This and #466 are going to clash a lot here. I commented there that I believe this copying will be fragile and potentially introduce bugs in the future. Why not just call `idToken.Claims` twice?

Reply:
Exactly, I wanted to do it the same way as #466 to reduce the conflicts. If you prefer to parse again, I have no problem with that.